pytorch_lightning/profiler/profilers.py

```diff
@@ -55,6 +41,38 @@ def start(self, action_name: str) -> None:
     def stop(self, action_name: str) -> None:
         """Defines how to record the duration once an action is complete."""
 
+    @abstractmethod
+    def summary(self) -> str:
+        """Create profiler summary in text format."""
+
+
+class BaseProfiler(AbstractProfiler, ABC):
+    """
+    If you wish to write a custom profiler, you should inherit from this class.
+    """
+
+    def __init__(self, local_rank: Optional[int] = None, log_dir: Optional[str] = None) -> None:
+        self.output_fname = getattr(self, "output_fname", None)
+        self.output_file = None
+        # the profiler can be used outside of lightning
+        # that's why we call `on_train_start` manually
+        self.on_train_start(local_rank=local_rank, log_dir=log_dir)
+
+    def on_train_start(self, local_rank: Optional[int] = None, log_dir: Optional[str] = None):
+        """
+        This function is used by the Trainer to inject local_rank with `DDP`
+        and `TensorBoardLogger` log_dir in the profiler.
+        """
+        self.local_rank = local_rank
+        self.log_dir = log_dir
+        self.prepare_file()
+
+    def prepare_file(self) -> None:
+        if self.output_fname:
+            fs = get_filesystem(self.output_fname)
+            self.output_file = fs.open(self.output_fname, "w")
+        self.write_streams = [self.output_file.write] if self.output_file else [log.info]
```

Comment:
is this meant to be called by other components? or overridden in subclasses? if not, could we mark the method as private?

Reply:
No, moved to private.